src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static final String VALID_PASSWORD = "qwerty";
+    private static final int VALID_AGE = 23;
+    private static final User VALID_LOGIN_1_USER = new User("gabriel",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User VALID_LOGIN_2_USER = new User("kate2006",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User INVALID_LOGIN_1_USER = new User("max",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User INVALID_LOGIN_2_USER = new User("",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User NULL_LOGIN_USER = new User(null,
+            VALID_PASSWORD, VALID_AGE);
+    private static final User INVALID_PASSWORD_1_USER = new User("Oleksandr",
+            "ghabf", VALID_AGE);
+    private static final User INVALID_PASSWORD_2_USER = new User("Volodymyr",
+            "abc", VALID_AGE);
+    private static final User INVALID_PASSWORD_3_USER = new User("Yevhen",
+            "", VALID_AGE);
+    private static final User VALID_PASSWORD_1_USER = new User("patrick",
+            "tyfhhfaa", VALID_AGE);
+    private static final User VALID_PASSWORD_2_USER = new User("Denius",
+            "lalakfjg", VALID_AGE);
+    private static final User NULL_PASSWORD_USER = new User("max",
+            null, VALID_AGE);
+    private static final User INVALID_AGE_1_USER = new User("Viktoriya",
+            VALID_PASSWORD, 16);
+    private static final User INVALID_AGE_2_USER = new User("max",
+            VALID_PASSWORD, -7);
+    private static final User VALID_AGE_1_USER = new User("engine768",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User VALID_AGE_2_USER = new User("car123456",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User NULL_USER = null;
+    private static final User VALID_USER = new User("Machine",
+            VALID_PASSWORD, VALID_AGE);
+    private static final User EXISTING_USER = new User("Machine",
+            VALID_PASSWORD, VALID_AGE);

```

Comment:
Just create some valid and invalid values for every field (ligin, password and age) and use it to create users